### DISCUSSION TITLE
WP-UT5 RTL8157 Link Speed issue with Synology DS1513+

### DISCUSSION THREAD

**Discussion by rsockanc** [bb-qq/r8152#433] [Q&A] [answered]
@wisdpi,  I just received my WP-UT5 RTL8157 today and have it connected to a Synology DS1513+ running DSM 7.1.1-42962 Update 6 (most current for this older version).  The install notes you have at https://www.wisdpi.com/blogs/news/how-to-use-usb-5gbps-ethernet-adapter-wp-ut5-on-synology-dsm work well and I am seeing the interface start and get an IP correctly even after a reboot.  The only issue is it seems it's only connecting at 1Gbps.  The switch side has a 10GBase-T SFP+ Transceiver that supports 10GBase-T SFP Copper - Support 1.25Gb/2.5Gb/5Gb/10Gb/s data rate and was tested with the WP-UT5 on a Windows pc to run at 5Gbps.  On the Synology, file copy test caps at 1Gbps and I cant see what the WP-UT5 is syncing at:  
sudo ethtool eth4 | grep Speed
Speed: Unknown!

|__8-1         0bda:8157:3000 00  3.20 5000MBit/s 544mA 1IF  (WisdPi USB 5G Ethernet 000334C8D6B10415)

4901.972688] usb 8-1: reset SuperSpeed USB device number 2 using etxhci_hcd-170202
[ 4902.685006] r8152 8-1:1.0 eth4: v2.18.1 (2024/05/20)
[ 4902.690073] r8152 8-1:1.0 eth4: This product is covered by one or more of the following patents:
                        US6,570,884, US6,115,776, and US6,327,625.

[ 4902.705821] r8152 8-1:1.0 eth4: chip rev 19
[ 4902.710125] usbcore: registered new interface driver r8152
[ 4902.885723] IPv6: ADDRCONF(NETDEV_UP): eth4: link is not ready
[ 4907.102920] IPv6: ADDRCONF(NETDEV_CHANGE): eth4: link becomes ready
[ 4907.110850] r8152 8-1:1.0 eth4: carrier on
[ 4909.716420] Synotify use 16384 event queue size

**Comment by wisdpi** [comment]
Looking at the logs the driver is already working properly, this speed shows that it's still currently a bug, but it doesn't affect it.
My guess from your situation is that it's the cable to switch part of the problem. Can you provide the switch model, sfp model, and cable performance? Ideally use CAT6A, short distance test CAT5A is fine but requires very good quality

**Reply by rsockanc** [reply]
Environment tested: Multiple new 1 meter CAT 6A cables, new Unifi 8 port aggregation switch, Wiitek 10Gb SFP+ to RJ45 IEEE 802.3az Compliant Module that is listed to support 1.25G/2.5G/5G/10GBase-T and also a 6COM 10GBase-T SFP+ Transceiver listed to support 1.25Gb/2.5Gb/5Gb/10Gb/s data rate.

The same combination of the Unifi switch, SFP+s, and network cables was tested with the WP-UT5 installed on a Windows desktop, and a 5Gbps link was auto-negotiated correctly and speed tested to 4+Gbps as expected.

However on the Synology with the WP-UT5 and corresponding driver shows: 

interface shows:
Settings for eth4:
        Supported ports: [ MII ]
        Supported link modes:   10baseT/Half 10baseT/Full 
                                100baseT/Half 100baseT/Full 
                                1000baseT/Full 
                                2500baseX/Full 
        Supported pause frame use: No
        Supports auto-negotiation: Yes
        Supported FEC modes: Not reported
        Advertised link modes:  10baseT/Half 10baseT/Full 
                                100baseT/Half 100baseT/Full 
                                1000baseT/Full 
                                2500baseX/Full 
        Advertised pause frame use: Symmetric Receive-only
        Advertised auto-negotiation: Yes
        Advertised FEC modes: Not reported
        Link partner advertised link modes:  10baseT/Full 
                                             100baseT/Full 
                                             1000baseT/Full 
        Link partner advertised pause frame use: No
        Link partner advertised auto-negotiation: Yes
        Link partner advertised FEC modes: Not reported
        Speed: Unknown!
        Duplex: Full
        Port: MII
        PHYAD: 32
        Transceiver: internal
        Auto-negotiation: on
Cannot get wake-on-lan settings: Operation not permitted
        Current message level: 0x00007fff (32767)
                               drv probe link timer ifdown ifup rx_err tx_err tx_queued intr tx_done rx_status pktdata hw wol
        Link detected: yes



When sent the command of: sudo ethtool -s eth4 speed 5000 duplex full autoneg off
result is 
Cannot set new settings: Invalid argument
  not setting speed
  not setting duplex
  not setting autoneg

**Reply by bb-qq** [reply]
Realtek devices require a special command to fix the link speed.
Please check the Readme: https://github.com/bb-qq/r8152/blob/master/ReadMe.txt#L19-L32

Also, from the information in this readme, it appears that the older kernels cannot handle 5gbps links. I am not sure about the actual speed, but I suspect that the display and configuration will not work.

**Reply by horch855** [reply]
@wisdpi @bb-qq I am struggling with the same behavior on DS920+.
I connected my WP-UT5 RTL8157 to a Synology DS920+ (DSM 7.2.1-69057 Update 5 ; Linux kernel: 4.4.302+) via CAT6 cable.
I'm not able to get a higher speed than 2.5Gbps. 
The special command for Realtek devices for 5Gbps speed doesn't work.

sudo ethtool -s eth2 autoneg on advertise 0x180000000002f
Cannot set new settings: Operation not supported
  not setting autoneg

The device is recognized by DMS as "USB 5G Ethernet - Realtek Semiconductor Corp.
My netgear switch GS110EMX recognized it as 5G too.

lsusb shows:
|__usb2          1d6b:0003:0404 09  3.00 5000MBit/s 0mA 1IF  (Linux 4.4.302+ xhci-hcd xHCI Host Controller 0000:00:15.0) hub
  |__2-2         0bda:8157:3000 00  3.20 5000MBit/s 544mA 1IF  (WisdPi USB 5G Ethernet 000334C8D6B1022F)

ethtool eth2
Settings for eth2:
        Supported ports: [ MII ]
        Supported link modes:   10baseT/Half 10baseT/Full
                                100baseT/Half 100baseT/Full
                                1000baseT/Full
                                2500baseX/Full
        Supported pause frame use: No
        Supports auto-negotiation: Yes
        Supported FEC modes: Not reported
        Advertised link modes:  10baseT/Half 10baseT/Full
                                100baseT/Half 100baseT/Full
                                1000baseT/Full
                                2500baseX/Full
        Advertised pause frame use: Symmetric Receive-only
        Advertised auto-negotiation: Yes
        Advertised FEC modes: Not reported
        Link partner advertised link modes:  100baseT/Full
                                             1000baseT/Full
        Link partner advertised pause frame use: No
        Link partner advertised auto-negotiation: Yes
        Link partner advertised FEC modes: Not reported
        Speed: Unknown!
        Duplex: Full
        Port: MII
        PHYAD: 32
        Transceiver: internal
        Auto-negotiation: on
Cannot get wake-on-lan settings: Operation not permitted
        Current message level: 0x00007fff (32767)
                               drv probe link timer ifdown ifup rx_err tx_err tx_queued intr tx_done rx_status pktdata hw wol
        Link detected: yes


dmesg:
[ 4982.650371] usb 2-2: reset SuperSpeed USB device number 2 using xhci_hcd
[ 4982.756398] r8152 2-2:1.0 eth2: v2.18.1 (2024/05/20)
[ 4982.761957] r8152 2-2:1.0 eth2: This product is covered by one or more of the following patents:
                        US6,570,884, US6,115,776, and US6,327,625.

[ 4982.779379] r8152 2-2:1.0 eth2: chip rev 19
[ 4982.784111] usbcore: registered new interface driver r8152
[ 4982.828181] IPv6: ADDRCONF(NETDEV_UP): eth2: link is not ready
[ 4987.180697] IPv6: ADDRCONF(NETDEV_CHANGE): eth2: link becomes ready
[ 4987.188046] r8152 2-2:1.0 eth2: carrier on


Is ther any way to activate the 5Gbps.

Thank you very much.

**Reply by wisdpi** [reply]
> @wisdpi @bb-qq I am struggling with the same behavior on DS920+.我在 DS920+ 上遇到同样的问题。 I connected my WP-UT5 RTL8157 to a Synology DS920+ (DSM 7.2.1-69057 Update 5 ; Linux kernel: 4.4.302+) via CAT6 cable.我通过 CAT6 电缆将 WP-UT5 RTL8157 连接到 Synology DS920+（DSM 7.2.1-69057 Update 5；Linux 内核：4.4.302+）。 I'm not able to get a higher speed than 2.5Gbps.我无法获得高于 2.5Gbps 的速度。 The special command for Realtek devices for 5Gbps speed doesn't work.针对 5Gbps 速度的 Realtek 设备的特殊命令不起作用。
> 
> sudo ethtool -s eth2 autoneg on advertise 0x180000000002fsudo ethtool -s eth2 autoneg 广告 0x180000000002f Cannot set new settings: Operation not supported无法设置新设置：不支持操作 not setting autoneg 不设置自动调整
> 
> The device is recognized by DMS as "USB 5G Ethernet - Realtek Semiconductor Corp.该设备被 DMS 认可为“USB 5G 以太网 - Realtek Semiconductor Corp.” My netgear switch GS110EMX recognized it as 5G too.我的netgear交换机GS110EMX也将其识别为5G。
> 
> lsusb shows: lsusb 显示： |__usb2 1d6b:0003:0404 09 3.00 5000MBit/s 0mA 1IF (Linux 4.4.302+ xhci-hcd xHCI Host Controller 0000:00:15.0) hub|__usb2 1d6b:0003:0404 09 3.00 5000MBit/s 0mA 1IF (Linux 4.4.302+ xhci-hcd xHCI 主机控制器 0000:00:15.0) 集线器 |__2-2 0bda:8157:3000 00 3.20 5000MBit/s 544mA 1IF (WisdPi USB 5G Ethernet 000334C8D6B1022F)|__2-2 0bda:8157:3000 00 3.20 5000MBit/s 544mA 1IF (WisdPi USB 5G 以太网 000334C8D6B1022F)
> 
> ethtool eth2 Settings for eth2: eth2 的设置： Supported ports: [ MII ]支持的端口：[ MII ] Supported link modes: 10baseT/Half 10baseT/Full支持的链接模式：10baseT/Half 10baseT/Full 100baseT/Half 100baseT/Full100baseT/半 100baseT/全 1000baseT/Full 1000baseT/完整 2500baseX/Full 2500baseX/完整 Supported pause frame use: No支持的暂停帧使用：否 Supports auto-negotiation: Yes支持自动协商：是 Supported FEC modes: Not reported支持的 FEC 模式：未报告 Advertised link modes: 10baseT/Half 10baseT/Full通告的链接模式：10baseT/半 10baseT/全 100baseT/Half 100baseT/Full100baseT/半 100baseT/全 1000baseT/Full 1000baseT/完整 2500baseX/Full 2500baseX/完整 Advertised pause frame use: Symmetric Receive-only广告暂停帧使用：仅对称接收 Advertised auto-negotiation: Yes广告自动协商：是 Advertised FEC modes: Not reported公布的 FEC 模式：未报告 Link partner advertised link modes: 100baseT/Full链接合作伙伴公布的链接模式：100baseT/Full 1000baseT/Full 1000baseT/完整 Link partner advertised pause frame use: No链接合作伙伴宣传的暂停帧使用：否 Link partner advertised auto-negotiation: Yes链接伙伴公布的自动协商：是 Link partner advertised FEC modes: Not reported链路合作伙伴公布的 FEC 模式：未报告 Speed: Unknown! 速度：未知！ Duplex: Full 复式: 全套 Port: MII 港口：信息产业部 PHYAD: 32 物理量：32 Transceiver: internal 收发器：内置 Auto-negotiation: on 自动协商：开启 Cannot get wake-on-lan settings: Operation not permitted无法获取 LAN 唤醒设置：不允许操作 Current message level: 0x00007fff (32767)当前消息级别：0x00007fff (32767) drv probe link timer ifdown ifup rx_err tx_err tx_queued intr tx_done rx_status pktdata hw woldrv 探测链路定时器 ifdown ifup rx_err tx_err tx_queued intr tx_done rx_status pktdata hw wol Link detected: yes 检测到链接：是
> 
> dmesg: 消息： [ 4982.650371] usb 2-2: reset SuperSpeed USB device number 2 using xhci_hcd[4982.650371] USB 2-2：使用xhci_hcd重置SuperSpeed USB设备编号2 [ 4982.756398] r8152 2-2:1.0 eth2: v2.18.1 (2024/05/20)[4982.756398] r8152 2-2:1.0 eth2:v2.18.1 (2024/05/20) [ 4982.761957] r8152 2-2:1.0 eth2: This product is covered by one or more of the following patents:[ 4982.761957] r8152 2-2:1.0 eth2：本产品受以下一项或多项专利保护： US6,570,884, US6,115,776, and US6,327,625.US6,570,884、US6,115,776 和 US6,327,625。
> 
> [ 4982.779379] r8152 2-2:1.0 eth2: chip rev 19[4982.779379]r8152 2-2:1.0 eth2:芯片版本19 [ 4982.784111] usbcore: registered new interface driver r8152[4982.784111]usbcore：注册了新的接口驱动程序r8152 [ 4982.828181] IPv6: ADDRCONF(NETDEV_UP): eth2: link is not ready[4982.828181]IPv6：ADDRCONF（NETDEV_UP）：eth2：链接未准备好 [ 4987.180697] IPv6: ADDRCONF(NETDEV_CHANGE): eth2: link becomes ready[4987.180697]IPv6：ADDRCONF（NETDEV_CHANGE）：eth2：链接准备就绪 [ 4987.188046] r8152 2-2:1.0 eth2: carrier on[4987.188046]r8152 2-2:1.0 eth2:运营商开启
> 
> Is ther any way to activate the 5Gbps.有什么办法可以激活5Gbps吗？
> 
> Thank you very much. 非常感谢。

Checking the logs, everything should be completely normal now. How did you conduct the speed test? If you're using read/write tests, make sure your disk array's read/write speed exceeds 2.5 Gbps. It's recommended to first use iperf3 for direct speed testing to ensure there are no issues with the link. Our test results show that the DS920+ with mechanical hard drives struggles to exceed a read/write speed of 2.5 Gbps; you'll need to switch to higher-performance solid-state drives to achieve that.

**Reply by horch855** [reply]
I think this is not a problem with the mechanical drives. Other users reported  over 300 to 400MB/s with mechanical drives in a 4 bay cluster.
It seems that the device could not be advised to 5G as ethhtool only shows 2500basexX/Full.
ethtool eth2
Settings for eth2:
        Supported ports: [ MII ]
        Supported link modes:   10baseT/Half 10baseT/Full
                                100baseT/Half 100baseT/Full
                                1000baseT/Full
                                2500baseX/Full
        Supported pause frame use: No
        Supports auto-negotiation: Yes
        Supported FEC modes: Not reported
        Advertised link modes:  10baseT/Half 10baseT/Full
                                100baseT/Half 100baseT/Full
                                1000baseT/Full
                                2500baseX/Full
        Advertised pause frame use: Symmetric Receive-only
        Advertised auto-negotiation: Yes

On DSM the network status is only: 2500 MBit/s, Vollduplex, MTU 9000
The special command for Realtek devices for 5Gbps speed doesn't work.
sudo ethtool -s eth2 autoneg on advertise 0x180000000002f
Cannot set new settings: Operation not supported
not setting autoneg

Is the any way to activate the 5G on the device ?

**Reply by bb-qq** [reply, marked as answer]
Performance problems have been reported on the Cedarview platform that are suspected to be caused by the USB host controller.
https://github.com/bb-qq/r8152/wiki/Compatibility

Since your model is also Cedarview platform, this may be related. This platform is too old...

**Comment by rsockanc** [comment]
@bb-qq , I just realized I posted this in general and not in issues, sorry.

**Reply by bb-qq** [reply]
This discussion is closed as it is a Cedarview platform issue. Please create a new issue in Issue if you need to discuss link speed issues, etc.

**Comment by beejj** [comment]
did this ever get fixed? I also have same issue on DS1019+ with WAVLINK 5Gbps RTL8157

**Comment by rsockanc** [comment]
I arrived at the same conclusion as well about the performance of the USB bus. Systems that are older with Cedarview and similar age devices will have performance issues. The best solution is to use 3x 1Gbps lines in a standard connection with no LACP/802.1X and just enabled SMB 3 multi channel on Synology file services. Collecting all four 1 Gbps interfaces will not provide any better performance.  In a raid configuration, you should see about 250 MBPs (~ 2.5 Gbps) when connected from a properly enabled Windows 10 or 11 computer with SMB three multi channel enabled which it is by default and a 10 GB nic. You could also attempt to have a client have three 1 gig nice also configured with multi channel and it may work, but only occasionally.
